File: java/lib/src/main/java/io/openweb3/walletpay/Transfer.java
package io.openweb3.walletpay;

import io.openweb3.walletpay.exceptions.ApiException;
import io.openweb3.walletpay.internal.api.TransferApi;
import io.openweb3.walletpay.models.TransferIn;
import io.openweb3.walletpay.models.TransferOut;

public final class Transfer {
	private final TransferApi api;

	Transfer() {
		api = new TransferApi();
	}

	public TransferOut create(final TransferIn transferIn) throws ApiException {
		return this.create(transferIn, new PostOptions());
	}

	public TransferOut create(final TransferIn transferIn, final PostOptions options) throws ApiException {
		try {
			return api.v1TransferCreate(transferIn).getData();
		} catch (io.openweb3.walletpay.internal.ApiException e) {
			throw Utils.WrapInternalApiException(e);
		}
	}
}

File: go/transfer.go
package walletpay

import (
	"context"

	"github.com/openweb3-io/wallet-pay-openapi/go/internal/openapi"
)

type (
	TransferIn  = openapi.TransferIn
	TransferOut = openapi.TransferOut
)

type Transfer struct {
	api *openapi.APIClient
}

func (e *Transfer) Create(ctx context.Context, transferIn *TransferIn) (*TransferOut, error) {
	req := e.api.TransferApi.V1TransferCreate(ctx)
	req = req.TransferIn(openapi.TransferIn(*transferIn))

	out, res, err := req.Execute()
	if err != nil {
		return nil, wrapError(err, res)
	}
	ret := TransferOut(out.Data)

	return &ret, nil
}
